### DISCUSSION TITLE
Is it possible to use @IgnoreIf to not build tests and completely disable them?

### DISCUSSION THREAD

**Discussion by achurch0607** [spockframework/spock#1836] [Q&A]
I have around 8000 tests in a suite that I am finally trying to update to the newest version of spock and geb. Moving from spock-core:1.3-groovy-2.5 to spock-core:2.3-groovy-4.0.
Using @IgnoreIf({functionToDetermineTrue()}) - I used to be able to run these tests and only tests that were not ignored would build and execute. I am now unable to just run a single tests because @IgnoreIf builds all of the tests not just the single test marked to run. Now all of the tests in my suite run and execute but are counted as ignored. After researching it seems to be a change from junit4 to junit5 that might be causing this. Any suggestions on how to disable tests from building so I can only run a single test that is marked to run from my master spreadsheet.

**Comment by szpak** [comment]
How do you execute them? With Idea, Gradle and Maven there is a way to point out just a single test to run and while most likely the other tests will be compiled in the first execution, in the subsequent run they should be treated as "not changed" and not compiled again (I'm not sure about Maven without Maven daemon).

Compilation is rather handled at the higher level than Spock.

**Reply by achurch0607** [reply]
Thanks for the quick reply, I executing with Gradle. I have run the tests alone but it still compiles all and marks them as ignored. Previous versions it would only compile and run the single test.  I am trying to identify where the breaking changes is that caused the @Ignoreif tests to start compiling every time instead of being completely disabled. This has caused a single test that should run < 1 min to take over 5 min to complete and build a report.

**Reply by szpak** [reply]
How to you call just one test from Gradle?

With [`--tests`](https://docs.gradle.org/current/userguide/java_testing.html#test_filtering)?
```
gw test --tests org.gradle.SomeSpec.oneFeature
```

And calling that twice triggers the whole `compilteTest` twice?

If yes, are you using Gradle daemon (but even without it, it should be cached)? And if yes again, please paste the output of the aforementioned command, called twice with `--console=verbose` (you can sanitize particular paths/packages/classes).

**Reply by szpak** [reply]
@achurch0607 Ping.

**Reply by achurch0607** [reply]
compileTest is not being triggered twice.  Running the single test with --tests does only build and run a single test. I would prefer to run the entire testuite useing  "gw test" and have the IgnoreIf feature not build ignored tests.

I have a spreadsheet with a list of tests. I use that to turn on and off tests as needed. By using @IgnoreIf({functionToDetermineTrue()}) My function that checks the spreadsheet to see if test should be executed. IgoreIf used to only build and run the tests that were marked as true (Now it is building them all and ignoring them). Do you have any suggestions on how to replace the ignoreif function with something similar that will only build and execute the desired tests?

**Reply by Vampire** [reply]
Unless you provide different proof that is very unlikely.
Even with Spock 1.x all tests were compiled, as it is `compileTestGroovy` that compiles all the tests.
`@IgnoreIf` always just ignored or skipped the tests at execution.
At compilation time it is impossible to evaluate the condition, so you could never conditionally compile.
It might be, that with Spock 1.x the tests were not **reported** as skipped and are now.
But the check of the condition was at runtime the same as now.
If you really want to control what is compiled, you probably need to manually configure the `compileTestGroovy` task with includes / excludes, so that only what you intend to is compiled.
That is nothing Spock can influence in any way.
